sound/soc/intel/boards/sof_sdw.c

```diff
@@ -225,13 +225,55 @@ int sdw_startup(struct snd_pcm_substream *substream)
 	return sdw_startup_stream(substream);
 }
 
+static int sdw_trigger(struct snd_pcm_substream *substream, int cmd)
+{
+	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
+	struct sdw_stream_runtime *sdw_stream;
+	struct snd_soc_dai *dai;
+	int ret;
+
+	/* Find stream from first CPU DAI */
+	dai = asoc_rtd_to_cpu(rtd, 0);
+
+	sdw_stream = snd_soc_dai_get_sdw_stream(dai, substream->stream);
+
+	if (!sdw_stream) {
```

Comment:
The !sdw_stream test doesn't look right. ``snd_soc_dai_get_sdw_stream`` will return ``ERR_PTR(-ENOTSUPP)`` if it can't get sdw_stream.

Reply:
fixed.